File: Tugas 1/Encapsulation.php
<?php

class Mobil {
    public $merk;
    public $tahunProduksi;
    public $warna;

    // Constructor
    public function __construct($merk, $tahunProduksi, $warna) {
        $this->merk = $merk;
        $this->tahunProduksi = $tahunProduksi;
        $this->warna = $warna;
    }

    // Getter untuk mendapatkan nilai property private
    public function getMerk() {
        return $this->merk;
    }

    public function getTahunProduksi() {
        return $this->tahunProduksi;
    }

    public function getWarna() {
        return $this->warna;
    }

    // Setter untuk mengubah nilai property private
    public function setMerk($merk) {
        $this->merk = $merk;
    }

    public function setTahunProduksi($tahunProduksi) {
        $this->tahunProduksi = $tahunProduksi;
    }

    public function setWarna($warna) {
        $this->warna = $warna;
    }
}

// Membuat objek Mobil
$mobil = new Mobil("Toyota", 2020, "Merah");

// Menggunakan getter untuk mendapatkan nilai property private
echo "Merk: " . $mobil->getMerk() . "<br>";
echo "Tahun Produksi: " . $mobil->getTahunProduksi() . "<br>";
echo "Warna: " . $mobil->getWarna() . "<br>";

// Menggunakan setter untuk mengubah nilai property private
$mobil->setMerk("Honda");
$mobil->setTahunProduksi(2022);
$mobil->setWarna("Biru");

// Menggunakan getter lagi untuk memastikan perubahan nilai property
echo "Merk: " . $mobil->getMerk() . "<br>";
echo "Tahun Produksi: " . $mobil->getTahunProduksi() . "<br>";
echo "Warna: " . $mobil->getWarna() . "<br>";

?>

File: Tugas 1/Encapsulation.py
class Mobil:
    def __init__(self, merk, tahun_produksi, warna):
        self.merk = merk
        self.tahun_produksi = tahun_produksi
        self.warna = warna

    def get_merk(self):
        return self.merk

    def get_tahun_produksi(self):
        return self.tahun_produksi

    def get_warna(self):
        return self.warna

    def set_merk(self, merk):
        self.merk = merk

    def set_tahun_produksi(self, tahun_produksi):
        self.tahun_produksi = tahun_produksi

    def set_warna(self, warna):
        self.warna = warna

# Membuat objek Mobil
mobil = Mobil("Toyota", 2020, "Merah")

print("Merk:", mobil.get_merk())
print("Tahun Produksi:", mobil.get_tahun_produksi())
print("Warna:", mobil.get_warna())

mobil.set_merk("Honda")
mobil.set_tahun_produksi(2022)
mobil.set_warna("Biru")

print("Merk:", mobil.get_merk())
print("Tahun Produksi:", mobil.get_tahun_produksi())
print("Warna:", mobil.get_warna())